src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/RolesController.kt

```diff
@@ -0,0 +1,85 @@
+package uk.gov.justice.digital.hmpps.manageusersapi.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.ArraySchema
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.manageusersapi.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.manageusersapi.service.AdminTypeReturn
+import uk.gov.justice.digital.hmpps.manageusersapi.service.RolesService
+
+@Validated
+@RestController
+class RolesController(
+  private val rolesService: RolesService
+) {
+
+  @PreAuthorize("hasRole('ROLE_ROLES_ADMIN')")
+  @Operation(
+    summary = "Get all audit events",
+    description = "Get all audit events, role required is ROLE_AUDIT",
+    security = [SecurityRequirement(name = "ROLE_ROLES_ADMIN")],
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Role Details Returned",
+        content = [
+          Content(
+            mediaType = "application/json",
+            array = ArraySchema(schema = Schema(implementation = Role::class))
```

Comment:
do you actually need this line, would have thought it should be calculated from the method signature?